rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
```

Comment:
This function is borrowed from [jQuery](https://api.jquery.com/jQuery.isEmptyObject/) and checks

> "properties on the object itself and properties inherited from prototypes (in that it doesn't use hasOwnProperty). The argument should always be a plain JavaScript Object as other types of object (DOM elements, primitive strings/numbers, host objects) may not give consistent results across browsers."

I added this one because the check also checks for the presence of a PlainObject.

But you're right since this is including a non-typecheck i drop that identifier.

- [ ] Remove `isEmptyObject` from `tcIdentifiers`